SOURCES/etc/xapi.d/plugins/updater.py

```diff
@@ -106,44 +106,30 @@ def display_package(p):
     return {'name': p.name, 'version': p.version, 'release': p.release, 'description': p.summary,
             'changelog': changelog, 'url': p.url, 'size': p.size, 'license': p.license}
 
-def build_repo_list(additional_repos):
+def build_repo_list(enabled_repos, additional_repos):
     repos = list(DEFAULT_REPOS)
     if additional_repos:
         repos += [x.strip() for x in additional_repos.split(',')]
-    return repos
+    return filter(lambda repo: repo in [x.id for x in enabled_repos], repos)
+
+def install_helper(session, args, mode):
+    assert mode == 'install' or mode == 'update'
```

Comment:
Just a suggestion:

```suggestion
    assert mode in ['install', 'update']
```

Reply:
Or `assert mode in ('install', 'update')` :p